cumulusci/tasks/bulkdata/tests/test_generatemapping.py

```diff
@@ -748,14 +758,187 @@ def test_split_dependencies__interviews_for_cycles(self, prompt):
+        )
+
+    @mock.patch("click.prompt")
+    @mock.patch("random.choice")
+    def test_split_dependencies__auto_pick_cycles_by_relationship_type(
+        self, random_choice, prompt
+    ):
+        t = _make_task(
+            GenerateMapping, {"options": {"path": "t", "break_cycles": "auto"}}
+        )
+
+        prompt.side_effect = AssertionError("Shouldn't be called")
+        random_choice.side_effect = AssertionError("Shouldn't be called")
+
+        split_dependencies = t._split_dependencies(
+            ["AccountLike__c", "ContactLike__c", "OpportunityLike__c", "Custom__c"],
+            {
+                # Primary_Contact__c is not nillable, so ContactLike__c must be loaded before
+                # AccountLike__c despite the cycle
+                "AccountLike__c": {
+                    "ContactLike__c": {
+                        "Primary_Contact__c": FieldData({"nillable": False})
+                    }
+                },
+                "ContactLike__c": {
+                    "AccountLike__c": {"AccountId": FieldData({"nillable": True})}
+                },
+                "OpportunityLike__c": {
+                    "AccountLike__c": {"AccountId": FieldData({"nillable": True})},
+                    "ContactLike__c": {
+                        "Primary_Contact__c": FieldData({"nillable": True})
                     },
                 },
+            },
+        )
+
+        self.assertEqual(
+            ["Custom__c", "ContactLike__c", "AccountLike__c", "OpportunityLike__c"],
+            split_dependencies,
+        )
+        random_choice.assert_not_called()
+
+    @mock.patch("click.prompt")
+    def test_split_dependencies__auto_pick_cycles(self, prompt):
+        t = _make_task(
+            GenerateMapping, {"options": {"path": "t", "break_cycles": "auto"}}
+        )
+
+        prompt.return_value = AssertionError("Shouldn't be called")
+
+        self.assertEqual(
+            set(["Custom__c", "Account", "Contact", "Opportunity"]),
+            set(
+                t._split_dependencies(
+                    ["Account", "Contact", "Opportunity", "Custom__c"],
+                    {
+                        "Account": {
+                            "Contact": {
+                                "Primary_Contact__c": FieldData({"nillable": True})
+                            }
+                        },
+                        "Contact": {
+                            "Account": {"AccountId": FieldData({"nillable": True})}
+                        },
+                        "Opportunity": {
+                            "Account": {"AccountId": FieldData({"nillable": True})},
+                            "Contact": {
+                                "Primary_Contact__c": FieldData({"nillable": True})
+                            },
+                        },
+                    },
+                )
             ),
         )
+
+
+@pytest.mark.skip()  # this is slow even with VCR
```

Comment:
I tried commenting this line to run the tests. The first 2 errored with:

```
E               vcr.errors.CannotOverwriteExistingCassetteException: Can't overwrite existing cassette ('/Users/dglick/Work/CumulusCI/cumulusci/tasks/bulkdata/tests/cassettes/TestIntegrationGenerateMapping.test_simple_generate.yaml') in your current record mode ('none').
E               No match for the request (<Request (GET) https://podname.salesforce.com/services/data/v48.0/sobjects>) was found.
E               No similar requests, that have not been played, found.
```

and the third failed with:

    TypeError: 'method' object is not subscriptable

What is the goal of these tests? Can it be achieved in a way that allows us to run them routinely so they don't end up broken and/or forgotten?

Reply:
They are integration tests. They actually run the task against a real org and ensure it doesn't crash or go into an infinite loop. They should be run routinely in CI but not necessarily on every commit that is pushed. Perhaps once a day per open PR? Scheduled github action?

In any case I added some pytest markers, command line flags, comments and documentation to make it all clearer.

Reply:
@prescod Thanks, this makes more sense now that they are identified as integration tests rather than skipped entirely.

Could you please follow up soon with a PR to set up the scheduled github action to run the integration tests regularly?

Reply:
https://github.com/SFDO-Tooling/CumulusCI/pull/1894

Not tested yet.